conan/tools/cmake/cmakedeps/cmakedeps.py

```diff
@@ -112,3 +114,35 @@ def _generate_files(self, require, dep, ret, find_module_mode):
         # file is common for the different configurations.
         if not os.path.exists(config.filename):
             ret[config.filename] = config.render()
+
+    def set_property(self, dep, prop, value):
+        self._properties[dep].update({prop: value})
```

Comment:
Using the bare dependency name might fail if there are 2 dependencies with same name as dependencies in the build and host context (``build_context_activated``) with same reference?